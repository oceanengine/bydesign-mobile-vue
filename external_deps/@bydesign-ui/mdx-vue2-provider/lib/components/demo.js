/**
 * Component Demo
 */
import createBEM from '../utils/bem';
import HL from './highlight';
import { openCodepen, openSandbox, copyToClipboard } from '../utils/editor';
var bem = createBEM('demo');

var Demo = function Demo() {
  return {
    props: {
      title: [String, Number],
      desc: [String, Number],
      code: String,
      src: String,
      srcRender: Function,
      maxHeight: Number
    },
    data: function data() {
      return {
        unfolded: false
      };
    },
    methods: {
      setUnfolded: function setUnfolded(value) {
        this.unfolded = value;
      }
    },
    render: function render() {
      var h = arguments[0];
      var title = this.title,
          desc = this.desc,
          code = this.code,
          srcRender = this.srcRender,
          unfolded = this.unfolded,
          setUnfolded = this.setUnfolded,
          _this$maxHeight = this.maxHeight,
          maxHeight = _this$maxHeight === void 0 ? 200 : _this$maxHeight;
      var containerStyle = {
        maxHeight: "".concat(maxHeight, "px")
      };
      return h("div", {
        "class": bem()
      }, [h("div", {
        "class": bem('title'),
        "attrs": {
          "id": title.toString()
        }
      }, [title]), srcRender !== undefined && h("div", {
        "class": bem('box')
      }, [h("div", {
        "class": bem('show')
      }, [srcRender()]), desc && h("div", {
        "class": bem('desc')
      }, [desc]), code !== undefined && h("div", {
        "class": bem('code-tool')
      }, [h("div", {
        "class": bem('operations')
      }, [h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openCodepen(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/codepen.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openSandbox(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/sandbox.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            copyToClipboard(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/copy.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            setUnfolded(!unfolded);
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/code.svg')
        }
      })])]), code !== undefined && unfolded && h("div", {
        "class": bem('code'),
        "style": containerStyle
      }, [h(HL, {
        "attrs": {
          "code": unescape(code)
        }
      })])])]);
    }
  };
};

export default Demo;