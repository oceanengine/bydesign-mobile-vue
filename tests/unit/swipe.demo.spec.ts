import Demo from '@src/components/swipe/demo/index.vue';
import { snapshotDemo } from '../demo';

snapshotDemo(Demo);
