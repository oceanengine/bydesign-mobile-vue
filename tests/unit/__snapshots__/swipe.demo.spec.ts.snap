// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo swipe">
  <div class="byted-header--wrapper">
    <div class="byted-header byted-header--default byted-header--fixed byted-hairline--bottom">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title"><span class="demo-header-icon"><img src="https://s2.pstatp.com/cdn/expire-1-d/byted-creative-app/static/ByDesign-logo.svg" class="img-icon"></span></div>
    </div>
  </div>
  <div><span class="byted-icon byted-icon-module-carousel" style="font-size: 20px;"><svg width="20" height="20" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M4 18C4 16.1144 4 15.1716 4.58579 14.5858C5.17157 14 6.11438 14 8 14H22H42H56C57.8856 14 58.8284 14 59.4142 14.5858C60 15.1716 60 16.1144 60 18V46C60 47.8856 60 48.8284 59.4142 49.4142C58.8284 50 57.8856 50 56 50H42H22H8C6.11438 50 5.17157 50 4.58579 49.4142C4 48.8284 4 47.8856 4 46V18Z" fill="transparent" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M48 28L52 32L48 36" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M16 28L12 32L16 36" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M23 43H25" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M31 43H33" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M39 43H41" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path></svg></span></div>
  <div>
    <div class="byted-swipe">
      <div class="byted-swipe__track" style="width: 0px; transition-duration: 0ms; transform: translate3d(0px, 0, 0);">
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">0</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">1</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">2</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">3</div>
        </div>
      </div>
      <div class="byted-swipe__indicators byted-swipe__indicators--default"><i class="byted-swipe__indicator byted-swipe__indicator--active"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="byted-swipe">
      <div class="byted-swipe__track" style="width: 0px; transition-duration: 0ms; transform: translate3d(0px, 0, 0);">
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">0</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">1</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">2</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">3</div>
        </div>
      </div>
      <div class="byted-swipe__indicators byted-swipe__indicators--default"><i class="byted-swipe__indicator byted-swipe__indicator--active"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="byted-swipe">
      <div class="byted-swipe__track" style="width: 0px; transition-duration: 0ms; transform: translate3d(0px, 0, 0);">
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">0</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">1</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">2</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">3</div>
        </div>
      </div>
      <div class="byted-swipe__indicators byted-swipe__indicators--default"><i class="byted-swipe__indicator byted-swipe__indicator--active"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="byted-swipe">
      <div class="byted-swipe__track" style="width: 0px; transition-duration: 0ms; transform: translate3d(0px, 0, 0);">
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">0</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">1</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">2</div>
        </div>
      </div>
      <div class="byted-swipe__indicators byted-swipe__indicators--default"><i class="byted-swipe__indicator byted-swipe__indicator--active"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="byted-swipe">
      <div class="byted-swipe__track" style="width: 0px; transition-duration: 0ms; transform: translate3d(0px, 0, 0);">
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">0</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">1</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">2</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">3</div>
        </div>
      </div>
      <div class="byted-swipe__indicators byted-swipe__indicators--default"><i class="byted-swipe__indicator byted-swipe__indicator--active" style="background-color: rgb(2, 120, 255);"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i></div>
    </div>
  </div>
  <div>
    <div class="byted-swipe">
      <div class="byted-swipe__track" style="width: 0px; transition-duration: 0ms; transform: translate3d(0px, 0, 0);">
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">0</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">1</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">2</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">3</div>
        </div>
      </div>
      <div class="custom-indicator">1/4</div>
    </div>
  </div>
  <div>
    <div class="byted-swipe">
      <div class="byted-swipe__track" style="width: 0px; transition-duration: 0ms; transform: translate3d(0px, 0, 0);">
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">0</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">1</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">2</div>
        </div>
        <div class="byted-swipe-item" style="width: 0px; transform: translate3d(0px, 0, 0);">
          <div class="show">3</div>
        </div>
      </div>
      <div class="byted-swipe__indicators byted-swipe__indicators--default"><i class="byted-swipe__indicator byted-swipe__indicator--active"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i><i class="byted-swipe__indicator"></i></div>
    </div>
    <div class="button-wrap">
      <div class="byted-button byted-button--primary byted-button--mini byted-button--plain byted-button--square byted-hairline--surround">&lt;</div>
      <div class="byted-button byted-button--primary byted-button--mini byted-button--plain byted-button--fade byted-button--square byted-hairline--surround">0</div>
      <div class="byted-button byted-button--primary byted-button--mini byted-button--plain byted-button--square byted-hairline--surround">1</div>
      <div class="byted-button byted-button--primary byted-button--mini byted-button--plain byted-button--square byted-hairline--surround">2</div>
      <div class="byted-button byted-button--primary byted-button--mini byted-button--plain byted-button--square byted-hairline--surround">3</div>
      <div class="byted-button byted-button--primary byted-button--mini byted-button--plain byted-button--square byted-hairline--surround">&gt;</div>
    </div>
  </div>
</div>
`;
